src/dut_power.rs

```diff
@@ -286,17 +286,23 @@ impl DutPwrThread {
         pwr_curr: AdcChannel,
         pwr_led: Arc<Topic<BlinkPattern>>,
     ) -> Result<Self> {
-        let pwr_line = find_line("DUT_PWR_EN").unwrap().request(
-            LineRequestFlags::OUTPUT,
-            1 - PWR_LINE_ASSERTED,
-            "tacd",
-        )?;
-
-        let discharge_line = find_line("DUT_PWR_DISCH").unwrap().request(
-            LineRequestFlags::OUTPUT,
-            DISCHARGE_LINE_ASSERTED,
-            "tacd",
-        )?;
+        let pwr_line = find_line("DUT_PWR_EN")
+            .ok_or_else(|| anyhow!("Could not find GPIO line DUT_PWR_EN"))?;
+        let discharge_line = find_line("DUT_PWR_DISCH")
+            .ok_or_else(|| anyhow!("Could not find GPIO line DUT_PWR_DISCH"))?;
+
+        // Early TACs have a i2c port expander on the power board.
+        // This port expander is output only and can not be configured as
+        // open drain.
+        // The outputs on later TACs should however be driven open drain
+        // for EMI reasons.
+        let flags = match pwr_line.chip().label() {
+            "pca9570" => LineRequestFlags::OUTPUT,
+            _ => LineRequestFlags::OUTPUT | LineRequestFlags::OPEN_DRAIN,
+        };
+
+        let pwr_line = pwr_line.request(flags, 1 - PWR_LINE_ASSERTED, "tacd")?;
+        let discharge_line = discharge_line.request(flags, DISCHARGE_LINE_ASSERTED, "tacd")?;
```

Comment:
Is this a critical error if the gpio lines can't be found here? Because the `tacd` will panic as the result is unwraped. If no we could just log an error condition here?

Reply:
> Is this a critical error if the gpio lines can't be found here?

Yes indeed. All TACs come equipped with a power board, so if we can find it there is likely something wrong with e.g. the current devicetree and this RAUC boot slot should be marked bad.
The idea is to let the `tacd` startup fail in all cases that indicate such an error to make sure that the RAUC slot will be eventually marked bad.